lib/eex/lib/eex/engine.ex

```diff
@@ -98,21 +101,24 @@ defmodule EEx.Engine do
   @doc """
   Returns an empty string as initial buffer.
   """
-  def init(_opts) do
+  @spec init(Keyword.t) :: <<>>
+  def init(opts) when is_list(opts) do
```

Comment:
I agree with not defining specs for a callback that is being defined in a __using__ macro, but.. in this case they are regular functions.
Shouldn't we still define the spec.
for example in this case, init/2 expects a Macro.t, but the function returns an empty string (a subset of Macro.t), Doesn't it help developers understand what the actual function is return, in contrast with what the callback expects to return?